File: backend/test-solutions/valid-parentheses.cpp
#include <stack>
#include <unordered_map>
using namespace std;

class Solution {
public:
    bool isValid(string s) {
        stack<char> stk;
        unordered_map<char, char> mapping = {
            {')', '('},
            {'}', '{'},
            {']', '['}
        };
        
        for (char c : s) {
            if (mapping.count(c)) {
                if (stk.empty() || stk.top() != mapping[c]) {
                    return false;
                }
                stk.pop();
            } else {
                stk.push(c);
            }
        }
        
        return stk.empty();
    }
};

File: backend/test-solutions/valid-parentheses.js
class Solution {
    isValid(s) {
        const stack = [];
        const mapping = {')': '(', '}': '{', ']': '['};
        
        for (const char of s) {
            if (char in mapping) {
                if (!stack.length || stack.pop() !== mapping[char]) {
                    return false;
                }
            } else {
                stack.push(char);
            }
        }
        
        return stack.length === 0;
    }
}